### ISSUE TITLE
failed trade

### ISSUE THREAD

**Issue by yononima** [bisq-network/bisq#4857]
### Description

An error occurred when taking the offer.

An error occurred at task: CreateTakerFeeTx
Exception message: Insufficient money,  missing 0.03586202 BTC

No funds have left your wallet yet.
Please try to restart your application and check your network connection to see if you can resolve the issue.Please try to restart your application and check your network connection to see if you can resolve the issue.

#### Device or machine

macbook pro, OS
<img width="564" alt="macOS Big Sur" src="https://user-images.githubusercontent.com/70004866/100391018-e39ebe80-303a-11eb-933e-ed416078cece.png">

#### Additional info

[bisq.log](https://github.com/bisq-network/bisq/files/5604939/bisq.log)

**Comment by ghost** [comment]
Documenting my investigation so far:

    Nov-24 18:11  Upgraded to 1.5.0

    Nov-26 @ 22:22 Bisq starts up; last seen block is height 658809
    Nov-26 @ 22:26 user sends OfferAvailabilityRequest for offer PR9AOEBS
    Bisq wallet is empty.  User funds trade directly from external wallet.
    Nov-26 @ 22:32 user funds trade wallet + 0.03804608 BTC
    Nov-26 @ 22:34 user funds trade wallet + 0.002 BTC
    Nov-26 @ 22:34 user sends OfferAvailabilityRequest for offer PR9AOEBS
    Nov-26 @ 22:35 user sends OfferAvailabilityRequest for offer PR9AOEBS
    Nov-26 @ 22:35 user takes offer but fails in CreateTakerFeeTx
        there is an exception thrown from BitcoinJ::completeTx -> presumably line 4170 within call to calculateFee
        Insufficient money,  missing 0.03586202 BTC
    Nov-26 @ 22:41 block 658810 is published

- it looks like the first UTXO was not considered for funding, why?
- there seems to be one extra call to OfferAvailabilityRequest which I don't see when trying to reproduce this.


@yononima if you can provide any extra clues as to what you did to make this happen, it might help.

**Comment by chimp1984** [comment]
@yononima Can you PM me (on keybase, same username) all logs files and a screenshot of the trade details and all tx Ids?
I assume the funds have been spent because for some reason the "reserved for offer" flag got removed so once one took the offer the reserved funds have not been on that address anymore. I saw similar cases but du lack of log could not find what causes it.

**Comment by chimp1984** [comment]
Did you use BTC or BSQ as trade fee?

**Comment by yononima** [comment]
I used BSQ.

I will send you the log files the soonest possible.

The error occurred when i sent funds from a external wallet to fund tx security. When funds transferred, i couldn't take the offer before the btc amount was 1 satoshi less. So i had to put more funds in order to proceed with the tx. When i added some more satoshi and tried to proceed with the tx, i got the error.

I couldn't understand why it was 1 satoshi less, as i scanned the QR for funding.

**Comment by chimp1984** [comment]
The 1 sat diff seems like a bug. thanks for your description was prob. some not well tested edge case. if we see the tx details we can see if the received funds was exact the requested ones. maybe the fee estimation was called again and the required funds changed (must not be the case). though if u cancel and take again same offer that is expected and required funds can be different as the fee is newly caclulated. I assume there is a bug that you got a new fee update, and while funding the trade the required funds got recalculated and failed then.

**Comment by chimp1984** [comment]
I tested with updating the fee frequently while creating or taking offers with/without external funding (also with multiple funding txs) but could not reporduce the bug. Remaining therory is that the fee estimation code has some issue with rounding, but have not looked into that yet. But all in call it seems to me a rare edge case bug not releated to the other more frequent bugs we got reported since 1.4.2.
